src/orion/core/worker/primary_algo.py

```diff
@@ -49,13 +48,18 @@ def seed_rng(self, seed):
     @property
     def state_dict(self):
         """Return a state dict that can be used to reset the state of the algorithm."""
-        return self.algorithm.state_dict
+        _state_dict = self.algorithm.state_dict
+        _state_dict.update(super(self.algorithm.__class__, self.algorithm).state_dict)
```

Comment:
Why aren't you using `self.algorithm.state_dict` instead?

Reply:
`self.algorithm.state_dict` only gets the child algorithm state but still need to sync _trials_info from base between copy algorithm and original algorithm. Although I did think the possibility to not sync the _trials_info, but it will change the original behavior.

Reply:
I'll mark this one as resolved as it's the same issue as discussed here: https://github.com/Epistimio/orion/pull/352#discussion_r389329855